docs/madr/decisions/049-policy-on-namespace.md

```diff
@@ -0,0 +1,681 @@
+
+```yaml
+# Producer policy, zone tag is not specified which means the policy has to be synced to all other zones,
+metadata:
+  namespace: backend-ns
+spec:
+  to:
+    - targetRef:
+        kind: MeshService
+        name: backend
+---
+# consumer policy, 'kuma.io/zone' tag is implicitly equal to the zone's name where policy is applied. 
+# No need to sync this policy to other zones.
+metadata:
+  namespace: frontend-ns
+spec:
+  to:
+    - targetRef:
+        kind: MeshService
+        name: backend
+        namespace: backend-ns
+```
+
+Zone-originated policies in `kuma-system` won't be synced to other zones.
+Zone-originated policies on Universal won't be synced to other zones as well.
+We might want to introduce namespaces to Universal in the future.
+
+### Order when merging
+
+By design, several policies can select the same DPP.
+In that case, policies are ordered and their `to` (or `from`) arrays are merged.
+Introducing namespace-scoped policies adds other steps to the policy comparison process.
+
+1. Is top-level `targetRef.kind` more specific (MeshServiceSubset > MeshService > MeshSubset > Mesh)?
+   When kinds are equal, go to the next step.
+2. Zone originated policy is more specific than global originated policy.
+   When policies originates at the same place, go to the next step.
+3. **[new step]** Compare namespaces so that `consumer-ns` > `producer-ns` > `kuma-system`.
+   If namespaces are equal, go to the next step.
+4. The policy is more specific if it's name is lexicographically less than other policy's name ("aaa" < "bbb" so that "
+   aaa" policy is more specific).
+
+### Cross-resource references
+
+When a policy in `zone-1` references a resource (MeshService or MeshHTTPRoute), 
+the reference should work after policy is synced to `zone-2`.
+
+Using `to[].targetRef.labels` is a reliable way to reference the resource, because labels are constant across zones. 
+For example, the label `kuma.io/display-name` has the same value no matter what zone or global we're using to read the resource. 
+That's why targetRefs like
+
+```yaml
+to:
+  - targetRef:
+      kind: MeshService
+      labels:
+        kuma.io/display-name: backend
+```
+are going to work even when policy is synced to another zone.
+The downside of this approach is that multiple MeshServices can be selected, 
+because we didn't specify `kuma.io/zone` and `k8s.kuma.io/namespace` labels.
+Also, using labels is a more verbose approach especially when it comes to referencing the resources in the same namespace.
+
+Another way to reference the resource is by using `name` and `namespace` fields:
+
+```yaml
+to:
+  - targetRef:
+      kind: MeshService
+      name: backend
+      namespace: backend-ns
+```
+
+but we have to implement "smart" de-referencing when the policy is synced to another zone.
+
+#### Considered Options
+
+* explicitly de-reference all `to[].targetRef.name` before syncing the resource to another zone
```

Comment:
So this will be in KDS in the originating zone?

Reply:
If we do this can't the system namespace be different depending on the zone?

Reply:
Added an implem section:

> #### Implementation
> 
> The update of `to[].targetRef` is going to happen in 2 steps:
> 
> 1. Global KDS client updates `to[].targetRef.name` and `to[].targetRef.namespace` when receives policies from the source zone
> 2. KDS client of the destination zone updates `to[].targetRef.namespace` when receives policies from Global
> 
> All clients should update `to[].targetRef.namespace` because this is an environment specific information 
> (i.e. system namespaces can be different, Universal deployments don't have namespaces at all).